app/models/position_history.rb

```diff
@@ -3,5 +3,39 @@ class PositionHistory < ApplicationRecord
   belongs_to :position
 
   validates :started_on, presence: true
-  validates :finished_on, presence: true
+  validate :last_position_finished, on: :create
+  validate :validate_position_history_overlap
+  validate :manager_exists
+
+  def last_position_finished
+    employee = Employee.find_by_id(employee_id)
+    position = employee.position_histories.last
+    if position.present? && position.finished_on.nil?
+      errors.add(:base, "The last position is active")
+    end
+  end
+
+  def validate_position_history_overlap
+    employee = Employee.find_by_id(employee_id)
+
+    if employee.position_histories.present? && !employee.position_histories.last.finished_on.nil?
+      employee.position_histories.each do |position|
+        position_interval = position.started_on..position.finished_on
+        if position_interval.cover?(started_on) || position_interval.cover?(finished_on)
+          errors.add(:base, "Position cannot overlap with other positions")
+        end
+      end
+    end
+  end
+
+  def manager_exists
+    department = employee.department
+
+    department.employees.each do |employee|
+      if employee.position_histories.last.present? && employee.position_histories.last.position.name == "manager"
```

Comment:
Create a constant MANAGER = 'manager' and use constant here. No magic numbers or strings.